dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,42 +52,203 @@ public Pattern getParameterPattern() {
         return paramRegex;
     }
 
-    public Map<String, DbParameterAccessor> getAllColumns(
-            final String tableOrViewName) throws SQLException {
-        String qry = "SELECT COLUMNNAME, COLUMNDATATYPE "
-                + "FROM SYS.SYSCOLUMNS WHERE REFERENCEID = "
-                + "(SELECT TABLEID FROM SYS.SYSTABLES WHERE TABLENAME = ?)";
-        return readIntoParams(tableOrViewName.toUpperCase(), qry);
+    @Override
+    public Map<String, DbParameterAccessor> getAllColumns(String tableName)
+            throws SQLException {
+        return new DatabaseTable(buildDatabaseObjectName(tableName)).getParams();
     }
 
-    private Map<String, DbParameterAccessor> readIntoParams(
-            String tableOrViewName, String query) throws SQLException {
-        try (PreparedStatement dc = getConnection().prepareStatement(query)) {
-            dc.setString(1, tableOrViewName);
+    private abstract class DatabaseObject {
+        protected DatabaseObjectName name;
+
+        private DatabaseObject(DatabaseObjectName objName) {
+            name = objName;
+        }
+
+        protected DatabaseMetaData getDbMetaData() throws SQLException {
+            return getConnection().getMetaData();
+        }
+
+        abstract protected ResultSet getObjectMetaData() throws SQLException;
 
-            ResultSet rs = dc.executeQuery();
+        public boolean exists() throws SQLException {
+            try (ResultSet rs = getObjectMetaData()) {
+                return rs.next();
+            }
+        }
+
+        abstract protected ResultSet getColumns() throws SQLException;
+
+        public Map<String, DbParameterAccessor> getParams() throws SQLException {
             Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
-            int position = 0;
-            while (rs.next()) {
-                String columnName = rs.getString(1);
-                String dataType = rs.getString(2);
-                DbParameterAccessor dbp = createDbParameterAccessor(
-                        columnName,
-                        Direction.INPUT,
-                        typeMapper.getJDBCSQLTypeForDBType(dataType),
-                        getJavaClass(dataType), position++);
-                allParams.put(NameNormaliser.normaliseName(columnName), dbp);
+            try (ResultSet rs = getColumns()) {
+                while (rs.next()) {
+                    String paramName = getColumnName(rs);
```

Comment:
Things look good so far. What do you think about moving the `getColumnName(rs)` inside `createColumnAccessor`? That way it will be closer to other similar activities. And here we can take the name from the accessor after creating it.